### PR TITLE
Refactor/bond amount

### LINKED ISSUE #924: use relaychain total bond amount calculate exchange rate (@alannotnerd )
**Context**:

The exchange rate depends on total staked currency in relay chain, and it is currently implemented by tracing reward/slash in relay chain, while reward/slash recorded by events, and it's impossible to refactor with xcm storage query.

**Suggestion**:
We should use total bonded amount instead. 
- Add `update_bonded_amount`
- Remove `record_staking_settlement`
- Stakeclient add update_bonded_amount.

### PR DESCRIPTION
**Pull request by alannotnerd**
Close #924 

- Remove `StakingPool` storage
- Remove `TotalSlash` storage
- Remove `record_staking_settlement` call
- Add `update_bond_amount` call to update exchange rate

### CODE REVIEW

File: pallets/liquid-staking/src/lib.rs [hunk 1]
```diff
@@ -434,6 +416,26 @@ pub mod pallet {
...
+            let exchange_rate = Rate::checked_from_rational(
+                bonded_amount,
+                T::Assets::total_issuance(Self::liquid_currency()?),
+            )
+            .ok_or(Error::<T>::InvalidExchangeRate)?;
+            let old_exchange_rate = Self::exchange_rate();
+            if exchange_rate > old_exchange_rate {
+                ExchangeRate::<T>::put(exchange_rate);
+                Self::deposit_event(Event::<T>::ExchangeRateUpdated(exchange_rate));
+            }
```
Comment (MrPai) [inline pallets/liquid-staking/src/lib.rs:435]: 1/3 
there may be a potential bug:
suppose when update_bonded_amount, in the meantime, someone unstake, so `T::Assets::total_issuance(Self::liquid_currency()?)` will decrease too.
so `new_exchange_rate` is still greater than old one.

File: pallets/liquid-staking/src/lib.rs [hunk 2]
```diff
@@ -470,12 +472,6 @@ pub mod pallet {
             }
 
             T::Assets::burn_from(Self::liquid_currency()?, &who, liquid_amount)?;
-            StakingPool::<T>::try_mutate(|b| -> DispatchResult {
-                *b = b
-                    .checked_sub(asset_amount)
-                    .ok_or(ArithmeticError::Underflow)?;
-                Ok(())
-            })?;
```
Comment (MrPai) [inline pallets/liquid-staking/src/lib.rs:478]: 2/3
now when user unstake, we only decrease `total_issuance of liquid_currency`, do not decrease `StakingPool` anymore.

File: pallets/liquid-staking/src/lib.rs [hunk 3]
```diff
@@ -489,50 +485,6 @@ pub mod pallet {
...
-                        *p = p.checked_add(amount).ok_or(ArithmeticError::Overflow)?;
-                        Ok(())
-                    })?;
-                    // update exchange rate.
-                    let exchange_rate = Rate::checked_from_rational(
-                        StakingPool::<T>::get(),
-                        T::Assets::total_issuance(Self::liquid_currency()?),
-                    )
-                    .ok_or(Error::<T>::InvalidExchangeRate)?;
-                    ExchangeRate::<T>::put(exchange_rate);
```
Comment (MrPai) [inline pallets/liquid-staking/src/lib.rs:519]: 3/3 
while in former logic, we use `StakingPool`, when user unstake, `StakingPool` will decrease too.

File: pallets/liquid-staking/src/lib.rs (hunk 1 above)
Comment (alannotnerd) [reply pallets/liquid-staking/src/lib.rs:435]: So do you think it's better to update exchange_ rate while doing `settlement`, I guess it makes more sense

File: pallets/liquid-staking/src/lib.rs [hunk 4]
```diff
@@ -607,19 +541,33 @@ pub mod pallet {
...
-                return Ok(().into());
+            let bond_extra = !bonded_amount.is_zero();
+
+            // Update exchange rate
+            let matching_pool = MatchingPool::<T>::get();
+            let exchange_rate = Rate::checked_from_rational(
+                bonded_amount + matching_pool.total_stake_amount,
+                T::Assets::total_issuance(Self::liquid_currency()?)
+                    + matching_pool.total_unstake_amount,
+            )
```
Comment (MrPai) [inline pallets/liquid-staking/src/lib.rs:565]: `matching_pool.total_unstake_amount` will be calculated by: **exchangeRate * xToken amount**

**Review by MrPai — CHANGES_REQUESTED** [review]

File: pallets/liquid-staking/src/lib.rs [hunk 5]
```diff
@@ -607,19 +541,35 @@ pub mod pallet {
...
+            let bond_extra = !bonded_amount.is_zero();
+
+            // Update exchange rate
+            let matching_pool = MatchingPool::<T>::get();
+            let old_exchange_rate = Self::exchange_rate();
+            let exchange_rate = Rate::checked_from_rational(
+                bonded_amount + matching_pool.total_stake_amount,
+                T::Assets::total_issuance(Self::liquid_currency()?)
+                    + old_exchange_rate
+                        .checked_mul_int(matching_pool.total_unstake_amount)
```
Comment (MrPai) [inline pallets/liquid-staking/src/lib.rs:566]: Should be 'total_unstake_amount/exchange_rate', let's fix it then we can merge

File: pallets/liquid-staking/src/lib.rs [hunk 6]
```diff
@@ -607,19 +541,35 @@ pub mod pallet {
...
 
-            if Self::matching_pool().is_matched() {
-                return Ok(().into());
+            let bond_extra = !bonded_amount.is_zero();
+
+            // Update exchange rate
+            let matching_pool = MatchingPool::<T>::get();
+            let old_exchange_rate = Self::exchange_rate();
+            let exchange_rate = Rate::checked_from_rational(
+                bonded_amount + matching_pool.total_stake_amount,
```
Comment (GopherJ) [inline pallets/liquid-staking/src/lib.rs:563]: maybe we should use `checked_add` instead

**Review by MrPai — APPROVED** [review]

**Pull request merged**